app/src/main/java/com/sildev/musicplayer/activity/MainActivity.kt

```diff
@@ -0,0 +1,290 @@
+package com.sildev.musicplayer.activity
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.media.MediaPlayer
+import android.os.Bundle
+import android.os.Handler
+import android.view.View
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.widget.SearchView
+import com.sildev.musicplayer.*
+import com.sildev.musicplayer.MusicPlayerHelper.getBitmapSong
+import com.sildev.musicplayer.adapter.SongAdapter
+import com.sildev.musicplayer.databinding.ActivityMainBinding
+import com.sildev.musicplayer.model.Song
+import com.sildev.musicplayer.presenter.MainContract
+import com.sildev.musicplayer.presenter.MainPresenter
+import com.sildev.musicplayer.service.PlaySongService
+import java.util.*
+
+class MainActivity : AppCompatActivity(), MainContract.View, SongAdapter.IClickSongItem,
+    SearchView.OnQueryTextListener {
+
+    private var backPressTime: Long = 0
+    private lateinit var controlMusicBottomSheetDialog: ControlMusicBottomSheetDialog
+    private lateinit var songAdapter: SongAdapter
+    private lateinit var songSearchView: SearchView
+    private val mainBinding: ActivityMainBinding by lazy {
+        ActivityMainBinding.inflate(
+            layoutInflater
+        )
+    }
+
+    private var currentPlayList: List<Song> = ArrayList()
```

Comment:
đã khởi tạo r thì ko cần khai báo kiểu 
hơn nữa dùng mutableList hoặc list của kotlin nhé 
```suggestion
    private var currentPlayList = ArrayList()
```